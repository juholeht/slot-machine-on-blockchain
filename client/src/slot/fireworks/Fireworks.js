import {
  React, useEffect, useState, useRef,
} from 'react';

import PropTypes from 'prop-types';
import Firework from './Firework';

const FIREWORK_COUNT = [...Array(15).keys()];

const createKeyForFirework = (i) => (`firework-${i}`);

const Fireworks = ({ fireworkDelay = 200 }) => {
  const willMount = useRef(true);

  const [fireworks, setFireworks] = useState([]);

  useEffect(() => {
  }, []);

  // ComponentWillMount Ã¡la hooks
  if (willMount.current) {
    FIREWORK_COUNT.map((_entry, i) => setTimeout(() => setFireworks((previousFireworks) => [
      ...previousFireworks, <Firework key={createKeyForFirework(i)} />], fireworkDelay * i)));
    willMount.current = false;
  }
  return fireworks;
};

Fireworks.propTypes = {
  fireworkDelay: PropTypes.number,
};

export default Fireworks;
