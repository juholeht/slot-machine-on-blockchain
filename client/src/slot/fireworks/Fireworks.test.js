import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Fireworks from './Fireworks';

configure({ adapter: new Adapter() });

describe('<Fireworks />', () => {
  it('renders properly with default values', () => {
    // FIXME: does not render fireworks which are initialized only after mount
    const wrapper = shallow(<Fireworks fireworkDelay={0} />);
    expect(wrapper).toMatchSnapshot();
  });
});
