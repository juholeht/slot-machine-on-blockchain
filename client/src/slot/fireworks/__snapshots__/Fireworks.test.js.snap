// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fireworks /> renders properly with default values 1`] = `null`;
